File: api/src/main/java/net/impactdev/gts/api/environment/Environment.java
package net.impactdev.gts.api.environment;

import com.google.common.collect.Maps;
import net.impactdev.gts.api.util.PrettyPrinter;

import java.util.Map;

public class Environment implements PrettyPrinter.IPrettyPrintable {

    private final Map<String, String> environment = Maps.newLinkedHashMap();

    public void append(String key, String version) {
        this.environment.put(key, version);
    }

    @Override
    public void print(PrettyPrinter printer) {
        for(Map.Entry<String, String> entry : this.environment.entrySet()) {
            printer.add("  * " + entry.getKey() + ": " + entry.getValue());
        }
    }

}

File: api/src/main/kotlin/net/impactdev/gts/api/environment/Environment.kt
package net.impactdev.gts.api.environment

import com.google.common.collect.Maps
import net.impactdev.gts.api.util.PrettyPrinter

class Environment : PrettyPrinter.IPrettyPrintable {
    private val environment: MutableMap<String, String> = Maps.newLinkedHashMap()
    fun append(key: String, version: String) {
        environment[key] = version
    }

    override fun print(printer: PrettyPrinter) {
        for ((key, value) in environment) {
            printer.add("  * $key: $value")
        }
    }
}